### PR TITLE
[Aichat] Remove chat completion api endpoint

### PR DESCRIPTION
**Pull request by fisher-alice**
This PR removes the `aichat_controller` `chat_completion` action as it was replaced by `start_chat_completion` and `chat_request` in https://github.com/code-dot-org/code-dot-org/pull/60703.

## Links
[jira](https://codedotorg.atlassian.net/browse/LABS-981)

## Testing story
Dashboard unit tests were updated.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/test/controllers/aichat_controller_test.rb [hunk 1]
```diff
@@ -223,71 +247,15 @@ class AichatControllerTest < ActionController::TestCase
     assert_equal chat_event2_response["chatMessageText"], chat_event2_stored["chatMessageText"]
   end
 
-  test 'authorized teacher has access to start_chat_completion' do
```
Comment (fisher-alice) [inline dashboard/test/controllers/aichat_controller_test.rb:226]: These were not removed by moved up in the test file.

**Review by cnbrenci — COMMENTED** [review]
LGTM, but a couple questions:

Did you mean #60391 in the pr description?

The jira task mentions:  `Use experiment flag ?enableExperiments=aichat-polling to test the new API and confirm functionality. Confirm with infra, and then enable in production by removing experiment flag.`

It looks like the experiment flag is still there- does that change to remove the experiment and fully enable the ActiveJob thing need to go in before this one?

**Comment by fisher-alice** [comment]
> LGTM, but a couple questions:
> 
> Did you mean #60391 in the pr description?
> 
> The jira task mentions: `Use experiment flag ?enableExperiments=aichat-polling to test the new API and confirm functionality. Confirm with infra, and then enable in production by removing experiment flag.`
> 
> It looks like the experiment flag is still there- does that change to remove the experiment and fully enable the ActiveJob thing need to go in before this one?

Oh yeah - sorry about that! ~~I'm proposing to merge this PR into https://github.com/code-dot-org/code-dot-org/pull/60703. Can't merge 60703 since `staging` is closed right now.~~ Thanks!

`staging` is open now so merged #60391.

**Comment by bencodeorg** [comment]
Should `chat_completion` also be removed from `ability.rb`?

File: dashboard/test/controllers/aichat_controller_test.rb [hunk 2]
```diff
@@ -79,59 +78,61 @@ class AichatControllerTest < ActionController::TestCase
...
 
-  test 'Bad request if storedMessages param is not included for chat_completion' do
+  test 'Bad request if storedMessages param is not included for start_chat_completion' do
     sign_in(@authorized_teacher1)
-    post :chat_completion, params: @missing_stored_messages_params, as: :json
+    post :start_chat_completion, params: @missing_stored_messages_params, as: :json
     assert_response :bad_request
   end
 
-  test 'returns user profanity status when chat message contains profanity' do
```
Comment (fisher-alice) [inline dashboard/test/controllers/aichat_controller_test.rb:107]: Unit tests to check that user and model profanity are detected are included in [`aichat_request_chat_completion_job_test`](https://github.com/code-dot-org/code-dot-org/blob/d712c4e1c19e60d8a66b4b2a457ec14db93d2353/dashboard/test/jobs/aichat_request_chat_completion_job_test.rb#L16-L43).

**Review by bencodeorg — APPROVED** [review]
🧹 LGTM!

**Pull request merged**